### PR TITLE
[CI] Implement skip-if-unchanged in drone

### PR DESCRIPTION
**Pull request by SwikritiT**
part of : 
https://github.com/owncloud/QA/issues/730

### CODE REVIEW

**Comment by SwikritiT** [comment]
## Confirmation that this works:
unit test changed: https://drone.owncloud.com/owncloud/activity/2558/13/5
acceptance tests changed https://drone.owncloud.com/owncloud/activity/2559/7/3
docs changed https://drone.owncloud.com/owncloud/activity/2563
no change https://drone.owncloud.com/owncloud/activity/2562

**Comment by phil-davis** [comment]
If I change some real code and a unit test, then the acceptance tests do not run.
See #1075 and https://drone.owncloud.com/owncloud/activity/2567

If I change some real test code only (do not touch any tests) then no unit tests and no acceptance tests run.
See #1076 and https://drone.owncloud.com/owncloud/activity/2568

If real code changes, then all code-style, code-analysis, unit and acceptance tests should run.

**Comment by SwikritiT** [comment]
> If I change some real code and a unit test, then the acceptance tests do not run. See #1075 and https://drone.owncloud.com/owncloud/activity/2567
> 
> If I change some real test code only (do not touch any tests) then no unit tests and no acceptance tests run. See #1076 and https://drone.owncloud.com/owncloud/activity/2568
> 
> If real code changes, then all code-style, code-analysis, unit and acceptance tests should run.

@phil-davis I've added new changes. Can you make sure that it works as desired?

File: .drone.star [hunk 1]
```diff
@@ -244,16 +245,17 @@ def codestyle(ctx):
...
-                        "commands": [
-                            "make test-php-style",
-                        ],
-                    },
-                ],
+                "steps": skipIfUnchanged(ctx, "lint") +
+                         [
+                             {
+                                 "name": "coding-standard",
+                                 "image": "owncloudci/php:%s" % phpVersion,
```
Comment (kiranparajuli589) [inline .drone.star:252]: use variable

Comment (SwikritiT) [reply .drone.star:252]: @sushmita56 PR https://github.com/owncloud/activity/pull/1073 does that change. So I don't think it's needed here

Comment (grgprarup) [reply .drone.star:252]: @SwikritiT IMO one whose PR gets merge at last must fix this.

**Comment by kiranparajuli589** [comment]
checked here https://github.com/owncloud/activity/pull/1075/files
looks good to me :+1:

**Comment by phil-davis** [comment]
I will squash and re-test with some combinations in PRs on top of this.

**Review by phil-davis — APPROVED** [review]

**Pull request merged**